### PR TITLE
PUB-1463 - Updated special characters

### PR DESCRIPTION
**Pull request by ChrisS1512**
### JIRA link ###

https://tools.hmcts.net/jira/browse/PUB-1463

### Change description ###

Fixed uploading of files with control characters.

Special characters (outside of the ascii 127) are stripped from the file name before being sent to the backend.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/controllers/ManualUploadController.ts [hunk 1]
```diff
@@ -43,9 +43,12 @@ export default class ManualUploadController {
       if (errors.fileErrors || errors.formErrors) {
         res.render('manual-upload', formValues);
       } else {
+        const originalFileName = req.file['originalname'];
+        const sanatisedFileName = fileHandlingService.sanitiseFileName(originalFileName);
```
Comment (KianKwa) [inline src/main/controllers/ManualUploadController.ts:47]: Typo. Should be 'sanitised' :)

File: src/main/service/fileHandlingService.ts [hunk 2]
```diff
@@ -33,6 +33,22 @@ export class FileHandlingService {
     return 'Please provide a file';
   }
 
+  /**
+   * Sanatises the filename of the file being uploaded before being sent to the backend.
```
Comment (KianKwa) [inline src/main/service/fileHandlingService.ts:37]: Another typo for 'sanitise'

File: src/main/service/fileHandlingService.ts [hunk 3]
```diff
@@ -50,23 +66,24 @@ export class FileHandlingService {
   /**
    * Stores an upload file into redis with an ID of the filename. It also removes the file from disk.
    * @param userId The user ID of the user uploading the file.
-   * @param fileName The filename of the file to store.
+   * @param originalFilename The filename before being sanatised.
+   * @param sanatisedFileName The filename of the file to store.
    */
-  async storeFileIntoRedis(userId, fileName) {
+  async storeFileIntoRedis(userId, originalFilename, sanatisedFileName) {
```
Comment (KianKwa) [inline src/main/service/fileHandlingService.ts:72]: Another 'sanatise' here

File: src/main/controllers/ManualUploadController.ts (hunk 1 above)
Comment (ChrisS1512) [reply src/main/controllers/ManualUploadController.ts]: Now fixed :)

File: src/main/service/fileHandlingService.ts (hunk 2 above)
Comment (ChrisS1512) [reply src/main/service/fileHandlingService.ts]: Now fixed :)

File: src/main/service/fileHandlingService.ts (hunk 3 above)
Comment (ChrisS1512) [reply src/main/service/fileHandlingService.ts]: Now merged :)

**Review by joshblackmoor — DISMISSED** [review]
LGTM 👍

**Review by furnivall — DISMISSED** [review]

**Review by KianKwa — APPROVED** [review]

**Review by furnivall — APPROVED** [review]

**Pull request merged**